### PR TITLE
Tw 2001: No previews for video

### PR DESCRIPTION
**Pull request by sherlockvn**
## Ticket
#2001 
## Root cause
- We accidentally upload the original file instead of thumbnail file
- We do not use correct file mimeType for thumbnail
## Solution
- Upload original file -> thumbnail file
- add mimeType for thumbnail when sending video

## Test recommendations
**Recommendations for how to test this, or anything else you are worried about?**
- Send video, image in multiple platform to see if have thumbnail or not
## Resolved
**Attach screenshots or videos demonstrating the changes**
- Android:

https://github.com/user-attachments/assets/2bb4eded-8918-451a-b559-45113884a9f0

- Web:

https://github.com/user-attachments/assets/d119b838-5c06-4ed0-828a-748a7c856334

- IOS:

### CODE REVIEW

**Comment by KhaledNjim** [comment]
Worked well for mp4 videos but webm is not displayed properly

image

**Comment by sherlockvn** [comment]
> Worked well for mp4 videos but webm is not displayed properly
> 
> image

this is expected behavior, because we can't display webp in the web platform

**Review by Te-Z — APPROVED** [review]

**Review by KhaledNjim — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]
LGTM

File: lib/presentation/extensions/send_file_web_extension.dart [hunk 1]
```diff
@@ -109,7 +110,10 @@ extension SendFileWebExtension on Room {
               .unsigned![fileSendingStatusKey] =
           FileSendingStatus.generatingThumbnail.name;
       await handleImageFakeSync(fakeImageEvent);
-      thumbnail ??= await generateVideoThumbnail(file);
+      thumbnail ??= await generateVideoThumbnail(
+        file,
+        uploadStreamController: uploadStreamController,
+      );
```
Comment (sherlockvn) [inline lib/presentation/extensions/send_file_web_extension.dart:116]: no need state

File: lib/presentation/extensions/send_file_extension.dart [hunk 2]
```diff
@@ -93,17 +93,15 @@ extension SendFileExtension on Room {
       }
 
       final formattedDateTime = DateTime.now().getFormattedCurrentDateTime();
-      final targetPath =
-          await File('${tempDir.path}/$formattedDateTime${fileInfo.fileName}')
-              .create();
-      fileInfo = fileInfo as ImageFileInfo;
+      final fileName = _generateThumbnailFileName(formattedDateTime, fileInfo);
+      final targetPath = await _getTargetParh(tempDir, fileName);
```
Comment (hoangdat) [inline lib/presentation/extensions/send_file_extension.dart:97]: it is a file, not a path. So, `_createThumbnailFile`

Comment (sherlockvn) [reply lib/presentation/extensions/send_file_extension.dart]: done

**Review by hoangdat — APPROVED** [review]

**Pull request merged**